File: Amina/Mahmood_PE10/SQLServerDatabase.java
/***
   *Amina Mahmood
   *ISTE 330.01-02
   *Mr. Floser
   *Database Connectivity and Access
***/
import java.sql.*;

public class SQLServerDatabase
{
   private String uri = "jdbc:mysql://localhost/bankdb?autoReconnect=true&useSSL=false&useUnicode=true&useJDBCCompliantTimezoneShift=true&useLegacyDatetimeCode=false&serverTimezone=UTC";
   private String driver = "com.mysql.cj.jdbc.Driver";
   private String user = "root";
   private String password = "student";
      
   Connection attributes = null;
      
   public SQLServerDatabase()
   {
      try 
      {
         Class.forName( driver );
         System.out.println("Driver loaded");
      }
      catch( ClassNotFoundException cnfe)
      {
         System.out.println("Connect find or load driver " + driver);
         cnfe.printStackTrace();
         System.exit(1);
      }
         
      if( connect() == false )
      {
         System.exit(1);
      }
   
      if( disconnect() == false )
      {
         System.exit(3);
      }
   }
   
   // Connect method - connect and then 
   public boolean connect()
   {
      try
      {
         attributes = DriverManager.getConnection(uri, user, password);
         System.out.println("Database Opens");
         return true;
      } 
      catch (SQLException e) 
      {
         System.out.println("Could not connect to Database " + uri);
         e.printStackTrace();
         System.exit(2);
         return false;
      }
   }

    // Close the database
   public boolean disconnect()
   {
      try 
      {
         if (attributes != null)
         {
            attributes.close();
            System.out.println("Database successfully closed.");
            return true;
         }
         else
         {
            return false; 
         }
      } 
      catch (SQLException sqle) 
      {
         System.out.println("Could not close database");
         sqle.printStackTrace();
         return false;
      }
   }
}

File: Amina/Mahmood_PE02/PE02_CSharp/SQLServerDatabase/SQLServerDatabase.cs
/***
   *Amina Mahmood
   *ISTE 330.01-02
   *Mr. Floser
   *Database Connectivity and Access
***/
using MySql.Data.MySqlClient;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MySQLDatabase
{
    class SQLServerDatabase
    {
        string mysqlConnStr = "server=127.0.0.1;Database=travel;uid=root;pwd=student;";

        MySqlConnection conn;

        public SQLServerDatabase()
        {
            try
            {
                if (connect() != false)
                {
                    // Open the database
                    Console.WriteLine("Database Opens");
                }

                // MySql close db
                close();
            }
            catch (MySqlException E)
            {
                Console.WriteLine("MySql Error: " + E.Message);
                Console.WriteLine("MySql " + E.StackTrace);
            }
        }

        // Connect method
        public Boolean connect()
        {
            try
            {
                conn = new MySqlConnection(mysqlConnStr);
                return true;
            }
            catch (MySqlException e)
            {
                Console.WriteLine("Could not connect to Database "+e);
                return false;
            }
        }
        
        // Close method
        public Boolean close()
        {
            try
            {
                if (conn != null)
                {
                    conn.Close();
                    Console.WriteLine("Database successfully closed.");
                    return true;
                }
                else
                {
                    return false;
                }
            }
            catch (MySqlException e)
            {
                Console.WriteLine("Could not connect to Database "+e);
                return false;
            }
        }
    }
}
